### PR TITLE
feat: Add TypeScript definitions for status flags

### PR DESCRIPTION
**Pull request by JakeGinnivan**
This PR adds the new status flags to the type definition as discriminated unions

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -358,6 +358,11 @@
   resolved "https://registry.yarnpkg.com/@babel/helper-validator-identifier/-/helper-validator-identifier-7.10.1.tgz#5770b0c1a826c4f53f5ede5e153163e0318e94b5"
   integrity sha512-5vW/JXLALhczRCWP0PnFDMCJAchlBvM7f4uk/jXritBnIa6E1KmqmtrS3yn1LAnxFBypQ3eneLuXjsnfQsgILw==
 
+"@babel/helper-validator-identifier@^7.10.3":
```
Comment (JakeGinnivan) [inline yarn.lock:361]: I just did a yarn locally and these were added, can revert if needed

**Comment by rrogowski** [comment]
@JakeGinnivan I just started using this library in a project and this was the first pain point I came across. Thanks for addressing this! 🎉

**Comment by orest22** [comment]
@JakeGinnivan Hey, thanks for PR something I was looking for. I also have a question do you think it's a good idea to make status an exportable type?
`export type Status = 'loading'| ect.'`

**Pull request merged**